### PR TITLE
#2689

### PR DESCRIPTION
**Pull request by vestrada**
https://github.com/craftercms/craftercms/issues/2689 - [studio-ui] S3datasource enhancements #2689

### CODE REVIEW

File: static-assets/components/cstudio-forms/data-sources/video-S3-upload.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ function(id, form, properties, constraints)  {
    	this.constraints = constraints;
    	
    	for(var i=0; i<properties.length; i++) {
+        if(properties[i].name == "repoPath") {
```
Comment (sumerjabri) [inline static-assets/components/cstudio-forms/data-sources/video-S3-upload.js:9]: Is this the right indentation?

File: static-assets/components/cstudio-forms/data-sources/video-S3-repo.js [hunk 2]
```diff
@@ -61,9 +61,9 @@ YAHOO.extend(CStudioForms.Datasources.VideoS3Repo, CStudioForms.CStudioFormDatas
 
     getSupportedProperties: function() {
         return [
-            { label: CMgs.format(langBundle, "profileId"), name: "profileId", type: "string" },
-            { label: CMgs.format(langBundle, "path"), name: "path", type: "string" },
-        ];
+            { label: CMgs.format(langBundle, "repositoryPath"), name: "path", type: "string" },
+            { label: CMgs.format(langBundle, "profileId"), name: "profileId", type: "string" }
+        ];s
```
Comment (sumerjabri) [inline static-assets/components/cstudio-forms/data-sources/video-S3-repo.js]: `;s`?

File: static-assets/components/cstudio-forms/data-sources/video-S3-upload.js [hunk 3]
```diff
@@ -72,6 +84,7 @@ YAHOO.extend(CStudioForms.Datasources.VideoS3Upload, CStudioForms.CStudioFormDat
 
 	getSupportedProperties: function() {
 		return [
+            { label: CMgs.format(langBundle, "repositoryPath"), name: "repoPath", type: "string" },
```
Comment (sumerjabri) [inline static-assets/components/cstudio-forms/data-sources/video-S3-upload.js:87]: Is this the right indentation?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-forms/data-sources/video-S3-repo.js (hunk 2 above)
Comment (vestrada) [reply static-assets/components/cstudio-forms/data-sources/video-S3-repo.js]: Done

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**